commands/audit/scarunner.go

```diff
@@ -53,18 +56,26 @@ func runScaScan(params *AuditParams, results *xrayutils.Results) (err error) {
 	log.Info(fmt.Sprintf("Preforming %d SCA scans:\n%s", len(scans), scanInfo))
 
 	defer func() {
-		// Make sure to return to the original working directory, executeScaScan may change it
+		// Make sure to return to the original working directory, building the dependency tree may change it
 		err = errors.Join(err, os.Chdir(currentWorkingDir))
 	}()
 	for _, scan := range scans {
-		// Run the scan
-		log.Info("Running SCA scan for", scan.Technology, "vulnerable dependencies in", scan.WorkingDirectory, "directory...")
-		if wdScanErr := executeScaScan(serverDetails, params, scan); wdScanErr != nil {
-			err = errors.Join(err, fmt.Errorf("audit command in '%s' failed:\n%s", scan.WorkingDirectory, wdScanErr.Error()))
+		// Get the dependency tree for the technology in the working directory.
+		treeResult, bdtErr := buildDependencyTree(scan, auditParams)
+		if bdtErr != nil {
+			err = errors.Join(err, fmt.Errorf("audit command in '%s' failed:\n%s", scan.WorkingDirectory, bdtErr.Error()))
 			continue
 		}
+		// Create sca scan task
+		auditParallelRunner.ScaScansWg.Add(1)
+		_, taskErr := auditParallelRunner.Runner.AddTaskWithError(executeScaScan(auditParallelRunner, serverDetails, auditParams, scan, *treeResult.FlatTree, treeResult.FullDepTrees), func(err error) {
```

Comment:
```suggestion
		_, taskErr := auditParallelRunner.Runner.AddTaskWithError(executeScaScan(auditParallelRunner, serverDetails, auditParams, scan, treeResult), func(err error) {
```
you can pass `treeResult` to reduce number of args